src/Engine/Component/Script.cpp

```diff
@@ -45,35 +45,41 @@ Json::Value Script::Save() const {
     return component;
 }
 
-void Script::FillPropertyMap() {
+void Script::AddToPropertyMap(std::string &name, int type, int size, void* data) {
 
-    int propertyCount = instance->GetPropertyCount();
+    if (!IsInPropertyMap(name, type))
+        propertyMap[name] = new Property(type, size, data);
 
-    for (int n = 0; n < propertyCount; n++) {
+}
+
+void Script::CopyDataFromPropertyMap(const std::string &name, void* target){
 
-        int typeId = instance->GetPropertyTypeId(n);
-        void *varPointer = instance->GetAddressOfProperty(n);
+    std::memcpy(target, propertyMap[name]->data, propertyMap[name]->size);
+    
+}
 
-        auto it = propertyMap.find(instance->GetPropertyName(n));
-        if (it != propertyMap.end())
-            if (propertyMap[instance->GetPropertyName(n)].first == typeId)
-                continue;
+void* Script::GetDataFromPropertyMap(const std::string &name){
 
-        int size = Managers().scriptManager->GetSizeOfASType(typeId, varPointer);
-        if (size != -1) {
+    return propertyMap[name]->data;
 
-            propertyMap[instance->GetPropertyName(n)] = std::pair<int, void*>(typeId, malloc(size));
-            memcpy(propertyMap[instance->GetPropertyName(n)].second, varPointer, size);
+}
+
+bool Script::IsInPropertyMap(const std::string &name, int type) {
+
+    auto it = propertyMap.find(name);
+
+    if (it != propertyMap.end() && it->second->typeID == type)
+        return true;
```

Comment:
Come to think of it, it would be even cleaner to just return the condition immediately.